### DISCUSSION TITLE
Field Keyword _ltxt

### DISCUSSION THREAD

**Discussion by CSWinnall** [cortexrd/Knack-Toolkit-Library#121] [Ideas]
Hi @cortexrd 

I have come up with a field keyword _ltxt (label text) which will be added to the field's description.  The idea being that we can keep our backend field titles short and replace the field title on the front end with the longer text, passed in as a param.  It works like this:

Field Title Backend:
GDPR

Field Description Backend:
`_ltxt=I give my consent for Noah's Place staff to have access to and process my information in adherence to GDPR 2018, with the purpose of delivering consistent and effective support.`

This will replace GDPR wherever the field appears on a form (could develop this idea further for details and tables) with the text after the =.

The problem:
If there are commas within the param, KTL will remove the space after it.  I understand why you do this.  Is there a way to escape the commas so that the space after the comma remains.

Craig

**Comment by cortexrd** [comment]
Hi Craig,

Great work on the future `_ltxt` keyword.  
Just a question: Wouldn't you prefer `_ttip` since it's already a well known naming convention?


Regarding your question about spaces:

Check out the `paramStr` version of the keyword.  It's there just for that.  It keeps the original version of the whole string in case you'd need it.  Just have to strip the unwanted portion(s).

![image](https://github.com/cortexrd/Knack-Toolkit-Library/assets/42851359/f2108649-79a7-4b93-9c5e-bf77ad947e55)

Otherwise, adding support for escaped characters would require additional development on the KTL, and I have very little time these days.  Actually, maybe it wouldn't even be required, now that you know about this.

ND

**Reply by CSWinnall** [reply]
Hi Norm

Thanks for your help this worked perfectly:

```
if (foundKwObj[fieldId]._ltxt && foundKwObj[fieldId]._ltxt[0].params.length) {
       //Get the Param String
       const param = foundKwObj[fieldId]._ltxt[0].paramStr;
       // remove the square brackets
       const labelTxt = param.substring(1, param.length - 1);
       if (view.type === 'form') {
               // change the label txt
               $('#' + view.key + ' #kn-input-' + fieldId + ' .kn-label span:not(.kn-required)').text(labelTxt);
       }   
}
```

Craig

**Comment by CSWinnall** [comment]
Great thanks, no the Param string will be just what I need.

The _ttip would do something else, we actually need the whole text on the
form. I am doing the tooltip I came up with this idea because of it.  You
maybe right to combine the 2 kw's using a parameter to make it work. At the
moment this solves a different issue for us.

Craig

**Comment by cortexrd** [comment]
I see you point and the benefit of this _ltxt keyword.

For example, if ever the text changes eventually for any reason, you only have to change it at one place, all all views will update at once, at run time.

A tooltip would only be a balloon that appears momentarily when you hover over a field, to give additional information.

I'd keep both keywords separate though, since they are quite different.

ND

**Comment by CSWinnall** [comment]
Ok great. I'd like to to extend this to details and grid headers.  The
backend is hard to navigate at the best of times without having really long
headers on the schema/tables.
